### DISCUSSION TITLE
What's the relationship between this project and KIC?

### DISCUSSION THREAD

**Discussion by spacewander** [Kong/kong-operator#423] [Q&A] [answered]
Both this project and kubernetes-ingress-controller reconcile Gateway API like Gateway & HTTPRoute. And even more, this project has a plan to distribute plugins: https://github.com/Kong/gateway-operator/issues/371. If people use KGO to distribute plugins, maybe the next step is to configure plugins with KGO... Otherwise, it's confusing that people need to use another controller to configure plugins.

So I have two questions:
* What's the relationship or boundary between this project and KIC?
* Why not implement the deployment feature inside the KIC?

**Comment by pmalek** [comment]
Hi @spacewander 👋 

KGO allows users to deploy and manage Kong (`DataPlane`) and KIC (`ControlPlane`) instances or use the [`Gateway`](https://gateway-api.sigs.k8s.io/api-types/gateway/) to deploy and manage those 2.

KIC reconciles all the Gateway API (and `Ingress`) resources in a non managed way (https://docs.konghq.com/kubernetes-ingress-controller/latest/gateway-api/#gatewayclass--gateway) which basically requires users to manage Kong instances themselves.

KIC currently is the only component that reconciles the "configuration" aspect of Kong, meaning it reconciles, `Service`s, `Ingress`es but also Gateway API `HTTPRoute`s, `GRPCRoute`s, `TLSRoute`s etc.

You can think of these 2 projects as:
- KGO can be used to deploy and manage KIC and Kong (think: an alternative to Helm)
- KIC is used to configure Kong

---

Plugin distribution is being implemented in KGO because KIC does not have (and does not plan to have) any management capabilities: it does not spawn or tear down resources. It "only" creates Kong configuration and applies it to Kong instance(s).

**Comment by spacewander** [comment]
@pmalek 
Thanks for your reply!

> KIC does not have (and does not plan to have) any management capabilities: it does not spawn or tear down resources. 

Could you explain why there is no plan to add management capabilities to KIC? For example, projects like Istio & Envoy Gateway provide configuring & management in the same binary, so that people don't need to maintain two binaries. What's the advantage of separating the management into a new project?

**Reply by pmalek** [reply, marked as answer]
When that becomes a reality the management **& configuration** responsibilities will land in KGO rather than in KIC.

There are no concrete plans for this but this is what will most likely happen at some point: KGO will be able to reconcile `HTTPRoute`s, `Ingress`es etc to create Kong configuration.

**Reply by spacewander** [reply]
@pmalek 
Thanks!
